src/content/blog/2023-09-20-tusd-200.md

```diff
@@ -0,0 +1,65 @@
+- Hooks can fully customize every HTTP response that is sent from tusd to the tus clients. This includes the status code, header fields, and the response body for success and error cases.
+- Hooks can reject an upload creation, for example if validation of the meta data fails or the user is not authorized.
+- Hooks can stop ongoing uploads to avoid unnecessary resource consumption, for example if an underlying resource is no longer available.
+- Hooks can customize the upload ID and its meta data when a new upload is created, influencing its storage destination.
+
+You can read more about this in the [hook system documentation](https://github.com/tus/tusd/blob/main/docs/hooks.md). If you are already using hooks, upgrading to tusd v2 is simple. The structure of the hook information has changed slightly, but still provides all the data as before. We encourage you to have a look at the documentation and the corresponding [examples](https://github.com/tus/tusd/tree/main/examples)!
+
+## Network resilience
+
+Tusd must accept large uploads even over unreliable network connections and, thus, must be resilient to various kinds of network issues and interruptions. With tusd v2, we worked on improvements in this area as well:
+
+A new **end-to-end** test suite has been developed that simulates various network conditions, such as slow upload speeds, dead connections, and concurrent requests. It allows us to observe tusd's behavior and asserts its correctness in as siutations. It has been central in our latest efforts to ensure that the tusd v2 release is ready for production.
+
+One topic that has been especially tricky in the past are concurrent upload requests. tusd does not support concurrent requests to the same upload resource, as parallel writes could trigger data loss. This rule is enforced through locks, which must be acquired before accessing or modifying an upload. While it solves the problem of concurrent access, it creates another one: When an ongoing upload is interrupted by network issues, the server might be unaware of this interruption and assume that the upload is still continuing and thus holding onto the upload lock. The client, on the other hand, might realize this interruption and abandon the previous upload request to resume the upload. This resumption would fail because the upload is still locked from the previous request, until the connection times out. Depending on the setup, such a timeout can be lengthy and thus cause annoyances for the end-users.
+
+As a solution, tusd v2 allows new requests to ask previous requests to the same upload resource to release their upload lock in a controlled and safe manner. This solves the problem of locked uploads during resumption, while also ensuring no data corruption. You can read more about the topic of upload locks in the [corresponding documentation](https://github.com/tus/tusd/blob/main/docs/locks.md).
+
+## Further improvements
```

Comment:
"Further improvements" sounds like a section with some quick small things but there are big features in there. I would maybe put this first, maybe as "highlights", including the two earlier points and only those two get a deeper dive afterwards.

Reply:
Interesting idea! I might want to try this for the new release post, but I fear that the current one is already getting quite large.